plugins/tableselection/plugin.js

```diff
@@ -889,6 +857,56 @@
 		setTimeout( function() {
 			editor.fire( 'afterPaste' );
 		}, 0 );
+
+		function shouldUseCustomPaste( selection, selectedCells ) {
+			var ranges = selection.getRanges(),
+				table = ranges.length && ranges[ 0 ]._getTableElement( { table: 1 } ),
+				isIgnoredTable = table && table.hasAttribute( ignoredTableAttribute ),
+				collapsedPartial = selectedCells.length === 1 && !rangeContainsTableElement( selection.getRanges()[ 0 ] ) && !boundarySelection,
+				boundaryWithoutTable = boundarySelection && !pastedTable;
+
+			// (#2945)
+			if ( isIgnoredTable ) {
+				return false;
+			}
+
+			var selectionAncestor = selection.getCommonAncestor(),
+				selectionExceedesTable = selectionAncestor &&
+					selectionAncestor.is &&
+					!selectionAncestor.is( 'table', 'tbody', 'tr', 'td' );
```

Comment:
`tableContent` was what I was thinking about, but `rangeContainsTableElement()` works just as fine and it reduces the complexity of that conditional, so I will go with it, thanks. As for `selection.isInTable()` it's not viable because it returns 'true' while pasting inside single table cell.